src/services/guidanceresponse.service.ts

```diff
@@ -17,14 +17,22 @@ module.exports.create = async (args: any, req: any) => {
   return await FhirUtilities.store(resource, GuidanceResponseModel, base_version);
 };
 
-const getMedicationCode = (medication: Medication | undefined) => {
+const getMedicationCode = (medication: Medication | MedicationRequest | undefined) => {
   // grab the medication drug code from the Medication resource
   let drugCode = null;
-  medication?.code?.coding?.forEach(medCode => {
-    if (medCode?.system?.endsWith('rxnorm')) {
-      drugCode = medCode?.code;
-    }
-  });
+  if (medication?.resourceType == 'Medication') {
+    medication?.code?.coding?.forEach(medCode => {
+      if (medCode?.system?.endsWith('rxnorm')) {
+        drugCode = medCode?.code;
+      }
+    });
+  } else {
+    medication?.medicationCodeableConcept?.coding?.forEach(medCode => {
```

Comment:
This does not handle the case with a medicationReference. Can you add code to handle if there is a medicationReference instead and you can expect that the Medication is a contained resource.